frontend/src/metabase/actions/containers/ActionCreator/QueryActionEditor.tsx

```diff
@@ -1,35 +1,22 @@
-import React, { useCallback } from "react";
+import React from "react";
 
 import NativeQueryEditor from "metabase/query_builder/components/NativeQueryEditor";
 
 import type NativeQuery from "metabase-lib/queries/NativeQuery";
-import type Question from "metabase-lib/Question";
-
-import { getTemplateTagParameters } from "metabase-lib/parameters/utils/template-tags";
 
 function QueryActionEditor({
-  question,
-  setQuestion,
+  query,
+  onChangeQuestionQuery,
 }: {
-  question: Question;
-  setQuestion: (q: Question) => void;
+  query: NativeQuery;
+  onChangeQuestionQuery: (query: NativeQuery) => void;
 }) {
-  const handleChange = useCallback(
```

Comment:
This logic has been pulled up to `<ActionCreator />`. [See the original discussion here](https://github.com/metabase/metabase/pull/27832#discussion_r1084677423).